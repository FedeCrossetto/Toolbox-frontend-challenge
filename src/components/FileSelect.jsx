import React from 'react';
import Form from 'react-bootstrap/Form';

const FileSelect = ({ list, selectedFile, setSelectedFile }) => {
  return (
    <Form.Select
      style={{ width: '90%', margin: 'auto' }}
      value={selectedFile}
      onChange={(e) => setSelectedFile(e.target.value)}
    >
      <option value="" hidden>
        Select a file
      </option>
      {list.map((file, index) => (
        <option key={index}>{file}</option>
      ))}
    </Form.Select>
  );
};

export default FileSelect;