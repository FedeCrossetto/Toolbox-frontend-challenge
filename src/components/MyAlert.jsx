import React from 'react';
import Alert from 'react-bootstrap/Alert';

const alertStyle = {
  width: '90%',
  margin: 'auto',
  marginTop: 24,
  opacity : '80%'
};

const MyAlert = ({ showAlert, setShowAlert }) => {
  return (
    showAlert && (
      <Alert
        variant="danger"
        style={alertStyle}
        onClose={() => setShowAlert(false)}
        dismissible
      >
        <p>The selected file does not contain information.</p>
      </Alert>
    )
  );
};

export default MyAlert;
