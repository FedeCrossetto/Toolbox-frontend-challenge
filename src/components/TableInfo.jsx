import React, { useEffect, useState } from 'react'
import Table from 'react-bootstrap/Table';
import Form from 'react-bootstrap/Form';

export const TableInfo = () => {

  const [list, SetList] = useState([]);
  const [files, SetFiles] = useState([]);
  const [selectedFile, setSelectedFile] = useState('');

  //Para obtener la lista de archivos 
  useEffect(() => {
    const getTest = () => {
      fetch('http://localhost:3000/files/list')
        .then(res => res.json())
        .then(res => SetList(res.files))
    }
    getTest()
  }, [])

  //Para obtener el archivo seleccionado de la lista.
  useEffect(() => {
    if (selectedFile) {
      SetFiles([])
      const regex = /\d+/;
      const getTest = () => {
        fetch(`http://localhost:3000/files/data?fileName=${selectedFile.match(regex)}`)
          .then(res => res.json())
          .then(res => SetFiles(res))
      }
      getTest()
      console.log('files', files)
    }
  }, [selectedFile])

  return (<>
    <Form.Select
      style={{ width: '90%', margin: 'auto' }}
      value={selectedFile}
      onChange={(e) => setSelectedFile(e.target.value)}
    >
      <option value="" hidden>
        Selecciona un archivo
      </option>
      {list.map((file, index) => (
        <option key={index}>{file}</option>
      ))}
    </Form.Select>
    <Table striped bordered hover size="sm" style={{ width: '90%', margin: 'auto', textAlign: 'center', marginTop: 24 }}>
      <thead>
        <tr style={{ textAlign: 'left' }}>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {(files.length != 0) ?
          files.slice(1).map((x, index) =>
          (
            <tr style={{ textAlign: 'left' }}
              key={index}>
              <td>{x.file}</td>
              <td>{x.text}</td>
              <td>{x.number}</td>
              <td>{x.hex}</td>
            </tr>
          )
          ) :
          <tr >
            <td colSpan="4">Table empty</td>
          </tr>
        }
      </tbody>
    </Table>
  </>

  );
}
