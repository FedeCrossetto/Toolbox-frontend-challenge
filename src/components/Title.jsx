import React from 'react'
import Container from 'react-bootstrap/Container'

const titleStyle = {
    backgroundColor: '#FF6666',
    paddingTop: 2,
    paddingBottom: 2,
    marginBottom: '16px'
};

const Title = () => {
    return (
        <>
            <Container  fluid style={titleStyle}>
                <h5 style={{ textAlign: 'left', color: 'white' }}>React Test App</h5>
            </Container>
        </>
    );
}
export default Title;