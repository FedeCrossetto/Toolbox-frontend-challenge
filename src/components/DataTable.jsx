import React, { useEffect, useState } from 'react';
import Table from 'react-bootstrap/Table';
import { getList, getFileData } from '../services/api';
import FileSelect from './FileSelect';
import MyAlert from './MyAlert';

const tableStyle = {
  width: '90%',
  margin: 'auto',
  textAlign: 'center',
  marginTop: 24
};

export const DataTable = () => {
  const [list, setList] = useState([]);
  const [files, setFiles] = useState([]);
  const [selectedFile, setSelectedFile] = useState('');
  const [showAlert, setShowAlert] = useState(false);

  useEffect(() => {
    setShowAlert(false);
    getList()
      .then((res) => setList(res.files))
      .catch((err) => {
        console.error(err);
        setShowAlert(true);
      });
  }, []);

  useEffect(() => {
    if (selectedFile) {
      setShowAlert(false);
      setFiles([]);
      getFileData(selectedFile)
        .then((res) => setFiles(res))
        .catch((err) => {
          console.error(err);
          setShowAlert(true);
        });
    }
  }, [selectedFile]);

  return (<>
    {<FileSelect
      selectedFile={selectedFile}
      list={list}
      setSelectedFile={setSelectedFile}
    />}
    <Table striped bordered hover size="sm" style={tableStyle}>
      <thead>
        <tr style={{ textAlign: 'left' }}>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {(files.length > 0) ?
          files.slice(1).map((x, index) =>
            <tr style={{ textAlign: 'left' }}
              key={index}>
              <td>{x.file}</td>
              <td>{x.text}</td>
              <td>{x.number}</td>
              <td>{x.hex}</td>
            </tr>
          ) :
          <tr >
            <td colSpan="4">Table empty</td>
          </tr>
        }
      </tbody>
    </Table>
    <MyAlert showAlert={showAlert} setShowAlert={setShowAlert} />
  </>
  );
}
