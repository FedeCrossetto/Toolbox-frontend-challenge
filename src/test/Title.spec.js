import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Title } from '../components/Title';


//Para comprobar que el componente se renderiza correctamente.
describe('Title component', () => {
  test('renders correctly', () => {
    const { container } = render(<Title />);
    expect(container).toMatchSnapshot();
  });
//Para comprobar que el componente muestra el texto 'React Test App'
  test('displays the title text', () => {
    const { getByText } = render(<Title />);
    expect(getByText('React Test App')).toBeInTheDocument();
  });
});