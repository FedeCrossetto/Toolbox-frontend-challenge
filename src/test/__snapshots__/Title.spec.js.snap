// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders correctly 1`] = `
<div>
  <div
    class="container-fluid"
    style="background-color: rgb(255, 102, 102); padding-top: 2px; padding-bottom: 2px; margin-bottom: 16px;"
  >
    <h5
      style="text-align: left; color: white;"
    >
      React Test App
    </h5>
  </div>
</div>
`;
